lib/image_spec/parser/gif.rb

```diff
@@ -20,5 +21,68 @@ def self.dimensions(stream)
   def self.size(stream)
     stream.size
   end
+  
+  #Heavily referenced from http://stackoverflow.com/a/7506880
+  #gcinfo because graphics control info
+  def self.gcinfo(stream)
+    #Skip header info because we know we are a gif
+    stream.seek(10, IO::SEEK_SET)
+    
+    #Skip some data and jump over the colour table
+    gct_flags = stream.read(1).unpack("C")[0]
+    
+    stream.seek(2, IO::SEEK_CUR)
+    
+    if gct_flags & 0x80 == 0x80
+        stream.seek(3 << ((gct_flags & 7) + 1), IO::SEEK_CUR)
+    end
+    
+    frames = 0
+    duration = 0
```

Comment:
suggestion to rename this `frame_delay` to make it clear that we are dealing with delay and not duration.